SE-XXX Typed throws.md

```diff
@@ -282,24 +303,113 @@ func userResultFromStrings(strings: [String]) throws GenericError -> User  {
 
 ```
 
-### Multiple layers of error flow
-
-`// TODO: Add examples and explain`
-
 ### Objective-C behaviour
 
+`// TODO: Not sure if we need that much history :D`
+
 In Objective-C, all current functions that take a double-pointer to `NSError` (a typical pattern in Foundation APIs) have an implicit type in the function signature, and, as has been pointed out, this does not provide more information that the current generic `Error`. But developers were free to subclass `NSError` and add it to its methods knowing that, the client would know at call time which kind of error would be raised if something went wrong.
 
+`// TODO: we should not pretend what is correct or the right way, there are always pros and cons, we should just summarize them`
+
 The assumption that every error in the current Apple's ecosystem was an `NSError` an hence, convertible to `Error`, made `throws` loose its type. But nowadays, there are APIs (`Decodable` -> `DecodingError`, `StringUTF8Validation` -> `UTF8ValidationError`, among others) that makes the correct use of the Swift's throwing pattern but the client (when those APIs are available), cannot distinguish one from the other one.
 
+`// TODO: Optionals have there place, I think we should just say that the are sometimes not expressive enough and don't support "go to catch" semantics`

```

Comment:
Here I wanted to say that developers many times prefer to use Optionals over throwing functions, not only for semantics, but because an Optional makes the code shorter and give the same few information than a throwing function (because of the lack of type)

Reply:
ah ok, this way round, i will put that in the todo